internal/manifests/queryfrontend/query_frontend.go

```diff
@@ -48,11 +48,11 @@ func BuildQueryFrontend(params manifestutils.Params) ([]client.Object, error) {
 
 	if gates.HTTPEncryption || gates.GRPCEncryption {
 		caBundleName := naming.SigningCABundleName(tempo.Name)
-		if err := manifestutils.ConfigureServiceCA(&d.Spec.Template.Spec, caBundleName, 0, 1); err != nil {
+		if err := manifestutils.ConfigureServiceCA(&d.Spec.Template.Spec, caBundleName, 0, 1, 2); err != nil {
```

Comment:
Does the `jaeger-query` container use the certificates?

Not strictly related to this PR, but could we use the container name instead of the index of the container in the deployment?
Because of using the index, the TLS volume mounting broke in the first place :| (because the `jaeger-query` container was added before the `tempo-query` container in the deployment spec).

Reply:
> Does the jaeger-query container use the certificates?

I assume no. But I will check.

> Not strictly related to this PR, but could we use the container name instead of the index of the container in the deployment?
Sure :)

Reply:
@andreasgerstmayr I have added a quick patch on top to determine the container index based on the container names.
In the long term, this functionality should probably be provided in `manifestutils`.